src/Database/Concerns/HasRelationships.php

```diff
@@ -135,7 +135,36 @@ public function hasRelation($name)
     public function getRelationDefinition($name)
     {
         if (($type = $this->getRelationType($name)) !== null) {
-            return (array) $this->{$type}[$name] + $this->getRelationDefaults($type);
+            return (array) $this->getRelationTypeDefinition($type, $name) + $this->getRelationDefaults($type);
+        }
+    }
+
+    /**
+     * Returns all defined relations of given type.
+     * @param string $type Relation type
+     * @return array
+     */
+    public function getRelationTypeDefinitions($type)
+    {
+        if (in_array($type, static::$relationTypes)) {
+            return $this->{$type};
+        }
+
+        return [];
+    }
+
+    /**
+     * Returns the given relation definition.
+     * @param string $type Relation type
+     * @param string $name Relation name
+     * @return array
```

Comment:
@Flynsarmy your tests indicate that a string can be returned here, if someone is using the short-hand version of defining a relation. It also would return `null` if the definition type or name specified did not exist.
```suggestion
     * @return array|string|null
```

Reply:
Could you also add a test case that shows an array being returned, just to tick that off as being tested?